### PR TITLE
feat: update README

### PR DESCRIPTION
**Pull request by alex-matthews**
### Changes

1. Reference to Cloudflare Tunnel removed from intro as it isn't a core part of the project and mentioning it in the second sentence of the README gives it undue emphasis. Reasons for its inclusion are given under FAQs.
2. Table of contents reintroduced given README length.
3. 'Overview' section added to emphasise that this is for home purposes, and to expand on the opinionated nature of the template.
4. 'Features' section added in place of 'Introduction' in order to better showcase the project.
5. Pre-start checklist migrated to 'General' and 'Infrastructure' prerequisites for clarity.
6. 'System requirements' subsection added under 'Machine preparation' heading to address some FAQs.
7. Moved all setup steps to 'Getting started' in order to separate those steps from instructions on how to use the project.
8. DNS: "Public" and "home" terminology adopted as it seems more apt than "external" and "internal" (e.g. "internal DNS" can mean internal to the cluster, the README talks about a "home" DNS server, etc).
9. DNS: Changed the internal-only subdomain used as a test from `hajimari` to `hubble` as the former is no longer a default app.
10. Added 'Usage: GitOps with Flux, Renovate and GitHub Actions' as a Level 2 heading to emphasise that this is the heart of the project. Reworded 'Renovatebot' subsection to give it more importance.
11. 'FAQs' section added.
12. Removed Flux SSH authentication instructions due to the reasons given as to why a public repo is recommended and the fact that the prerequisites specify a public repo. Determined users can figure this out for themselves.

#### Major

1. 'Usage'->'Flux' not yet documented. Proposed content: Provide an overview of how Flux works in the context of the project, i.e. directory structure, kustomizations and "fluxtomizations", dependencies etc. A bit like [this](https://jjgadgets.tech/pages/flux-repo-structure) but more high level. And maybe a brief how-to on routine cluster tasks that can be done using Git. 
2. 'Usage'->'GitHub Actions' not yet documented. Proposed content: Unknown, although tools such as `actions-runner-controller` and `flux-diff` could be mentioned. Custom bots too.
3. 'What's next'->'Addons' only partially documented. Proposed content: Advice on each addon (whether in the README or in `addons.yaml`). Advice meaning a one-line description of what the app does and a one-line advisory on when to enable it or not (like done for `grafana` and `kube-prometheus-stack`).
4. More FAQs need to be added. One potential candidate (which may need to be broken down into several Q&As): _"Working with Ansible after initial cluster deployment."_. Come to think of it, Ansible probably deserves its own subsection under 'Usage'.

#### Minor

1. `task brew:deps` installs more CLI tools than required (this is possibly intentional)
2. Debian installation instructions for RPi refer back to steps 3 and 4 of the installation instructions for AMD64, which assume a non-root user. Proposed fix: Include adduser instructions for RPi.
3. Step 3 of the Debian installation instructions for AMD64 needs adjustment for RPi because the command `apt install -y sudo` doesn't work until `apt update && apt upgrade` is run. Proposed fix: Make a note of this peculiarity.
4. `flux check --pre` seems redundant. Proposed fix: Remove flux check step.
5. SSL certificates: Instructions for switching to the production servers results in all secrets being re-encrypted, introducing many unnecessary changes that then get pushed to GitHub (unless the user realizes and intervenes). Proposed fix: Unknown.
6. `csi-driver-nfs` is an addon yet it is also listed under 'What's next'->'Storage' as a project that might be worth checking out. Proposed fix: Explain `csi-driver-nfs` in more detail (whether in the README or in `addons.yaml`) and remove it from the list of storage projects.

### Ideas

1. Regarding the DNS server prerequisite: Consider whether to offer (a) an in-cluster solution such as `blocky`, and/or (b) advice on how to accomplish split DNS using NextDNS.
2. Regarding Cloudflare: Cloudflare domain settings used to be handled by Terraform. People who are unfamiliar with Cloudflare won't necessarily know the correct settings to use (e.g. `SSL mode=strict`) and might do something that goes against best practice or that breaks things. Consider providing instructions on Cloudflare settings (or reintroducing Terraform, although Terraform probably adds unnecessary complexity).
3. Regarding 'What's next': Suggest offering a few more addons or community solutions on how to extend the project, as has been done for storage. Core features could be listed here (e.g. replicated storage, backups, authentication, VPN, remote management etc) and the recommended solution put next to each, e.g. "replicated storage: `rook-ceph` is recommended if your nodes have a dedicated data disk and sufficient memory, otherwise `longhorn` is a safe bet". Slightly reductive, yet helpful for beginners. Simplify comparisons by stating pros and cons briefly and offering a reference implementation for each approach, e.g. "authentication: self-hosted vs cloud, reference implementations being (self-hosted) `authelia` w/ `lldap` and (cloud) Cloudflare Zero Trust w/ GitHub as OIDC provider".

### CODE REVIEW

**Comment by onedr0p** [comment]
No need to include drafting notes directly in the README. You can post your comments on why you removed them here in your PR description.

**Comment by alex-matthews** [comment]
Thanks, I'm keen to deliver a finished product. It would be good to know whether I'm on the right track with respect to the matters I have identified as unresolved issues.

**Comment by onedr0p** [comment]
I'll give this a closer look this weekend when I'm back home.

File: README.md [hunk 1]
```diff
@@ -336,53 +393,58 @@ Lets get the required workstation tools installed and configured.
...
 
-If you're having trouble with DNS be sure to check out these two Github discussions, [Internal DNS](https://github.com/onedr0p/flux-cluster-template/discussions/719) and [Pod DNS resolution broken](https://github.com/onedr0p/flux-cluster-template/discussions/635).
+   ```sh
+   # /etc/dnsmasq.d/99-k8s-gateway-forward.conf
+   server=/${bootstrap_cloudflare_domain}/${bootstrap_k8s_gateway_addr}
+   ```
+
+2. Restart dnsmasq on the server.
+
+3. Query an internal-only subdomain from your workstation: `dig @${home-dns-server-ip} hubble.${bootstrap_cloudflare_domain}`. It should resolve to `${bootstrap_internal_nginx_addr}`.
```
Comment (onedr0p) [inline README.md:435]:
```suggestion
3. Query an internal-only subdomain from your workstation: `dig @${home-dns-server-ip} echo-server.${bootstrap_cloudflare_domain}`. It should resolve to `${bootstrap_internal_nginx_addr}`.
```

**Pull request closed without merging**